File: src/main/java/me/arasple/mc/trmenu/action/acts/ActionClose.java
package me.arasple.mc.trmenu.action.acts;

import me.arasple.mc.trmenu.TrMenu;
import me.arasple.mc.trmenu.action.base.AbstractAction;
import org.bukkit.Bukkit;
import org.bukkit.entity.Player;

/**
 * @author Arasple
 * @date 2019/12/28 19:07
 */
public class ActionClose extends AbstractAction {

    @Override
    public String getName() {
        return "close|shut";
    }

    @Override
    public void onExecute(Player player) {
        Bukkit.getScheduler().runTask(TrMenu.getPlugin(), player::closeInventory);
    }

}

File: src/main/kotlin/menu/api/action/impl/ActionClose.kt
package menu.api.action.impl

import org.bukkit.entity.Player
import trmenu.api.action.InternalAction
import trmenu.api.action.base.AbstractAction
import trmenu.api.action.base.ActionDesc
import trmenu.api.action.base.ActionOption
import trmenu.module.display.session

/**
 * @author Arasple
 * @date 2021/1/29 22:25
 */
class ActionClose : InternalAction() {

    override fun onExecute(player: Player, placeholderPlayer: Player) {
        player.session().close(closePacket = true, updateInventory = true)
    }

    companion object : ActionDesc {

        override val name = "close|shut".toRegex()

        override val parser: (Any, ActionOption) -> AbstractAction = { _, _ ->
            ActionClose()
        }

    }

}